test/tools/syntaxTestTool.cpp

```diff
@@ -0,0 +1,354 @@
+	else
+		for (auto const &expectation: m_test.expectations)
+		{
+			cout << (expectation.type == "Warning" ? color::yellow() : color::red())
+				 << "    " << expectation.type << ": " << expectation.message
+				 << color::reset() << endl;
+		}
+}
+
+void SyntaxTestTool::printErrorList(ErrorList const& _errorList, bool const _parserError) const
+{
+	if (_errorList.empty())
+		cout << color::green() << "    Success" << color::reset() << endl;
+	else
+		for (auto &error: _errorList)
+		{
+			if (error->typeName() == "Warning")
+			{
+				if (_parserError) continue;
+				cout << color::yellow();
+			}
+			else
+				cout << color::red();
+
+			cout << "    ";
+			if (_parserError)
+			{
+				int line = getLineNumber(
+					m_test.source,
+					boost::get_error_info<errinfo_sourceLocation>(*error)->start
+				);
+				if (line >= 0)
+					cout << "(" << line << "): ";
+			}
+			cout << error->typeName() << ": " << errorMessage(*error) << color::reset() << endl;
+		}
+}
+
+bool SyntaxTestTool::process()
+{
+	bool match = true;
+	bool parserError = false;
+	ErrorList errorList;
+
+	m_test = SyntaxTestParser().parse(m_path.string());
+
+	try
+	{
+		errorList = parseAnalyseAndReturnError(m_test.source, true, true, true).second;
+
+		if (errorList.size() != m_test.expectations.size())
+			match = false;
+		else
+		{
+			for (size_t i = 0; i < errorList.size(); i++)
+			{
+				if (!(errorList[i]->typeName() == m_test.expectations[i].type) ||
+					!(errorMessage(*errorList[i]) == m_test.expectations[i].message))
+				{
+					match = false;
+					break;
+				}
+			}
+		}
+	}
+	catch (...)
+	{
+		match = false;
+		parserError = true;
+		errorList = m_compiler.errors();
+	}
+
+	cout << color::bold() << m_name << ": " << color::reset();
+
+	if (match)
+		cout << color::green() << "OK" << color::reset() << endl;
+	else
+	{
+		cout << color::red() << "FAILED" << color::reset() << endl;
+
+		cout << "  Contract:" << endl;
+		printContract();
+
+		if (parserError)
+		{
+			cout << "  " << color::inverse() << color::red() << "Parsing failed:" << color::reset() << endl;
+			printErrorList(errorList, true);
+		}
+		else
+		{
+			cout << "  Expected result:" << endl;
+			printExpected();
+			cout << "  Obtained result:" << endl;
+			printErrorList(errorList, false);
+		}
+		cout << endl;
+
+		if (parserError)
+			cout << "(e)dit/(s)kip/(q)uit? ";
+		else
+			cout << "(e)dit/(r)eplace/(s)kip/(q)uit? ";
+		cout.flush();
+
+		bool done = false;
+		while (!done)
+		{
+			switch(cin.get())
+			{
+				case 's':
+					done = true;
+					break;
+				case 'r':
+					if (!parserError)
+					{
+						ofstream file(m_path.string(), ios::trunc);
+
+						file << m_test.source;
+						file << "// ----" << endl;
+						for (auto &error: errorList)
+							file << "// " << error->typeName() << ": " << errorMessage(*error) << endl;
+
+						done = true;
+					}
+					break;
+				case 'e':
+					cout << endl << endl;
+					(void)system((editor + " " + m_path.string()).c_str());
```

Comment:
Any coding style preferences for ignoring return values as in this case?

Reply:
I'm not aware of our complier settings reporting this...